src/evox/metrics/igd.py

```diff
@@ -16,5 +16,9 @@ def igd(objs: torch.Tensor, pf: torch.Tensor, p: int = 1):
     :note:
         The IGD score is lower when the approximation is closer to the Pareto front.
     """
-    min_dis = torch.cdist(pf, objs).min(dim=1).values
+    nan_idx = torch.any(torch.isnan(objs), dim=1)
```

Comment:
Revert these changes. IGD doesn't need to handle NaN values, the algorithms should.